crates/bevy_transform/src/hierarchy/named_hierarchy.rs

```diff
@@ -0,0 +1,505 @@
+use bevy_core::Name;
+use bevy_ecs::prelude::*;
+//use serde::{Deserialize, Serialize};
+use private::*;
+use smallvec::{smallvec, SmallVec};
+
+use crate::{Children, Parent};
+
+// TODO: Panic if the root entity is not defined prior to a query
+// TODO: Create a simplified version without the children ?!
+// TODO: Should I just use usize? and get done with it?
+
+/// Provides a way of describing a hierarchy or named entities
+/// and means for finding then in the world
+///
+/// By default and to save memory the nodes are indexed using `u16`
+/// but you can change it when needed.
+#[derive(Debug, Clone)]
+pub struct NamedHierarchy<I: Index = u16> {
+    /// Entity identification made by parent index and name
+    entities: Vec<(I, Name)>,
+    // ? NOTE: SmallVec<[u16; 10]> occupy the same 32 bytes as the SmallVec<[u16; 8]>, but the latter
+    // ? should be only take 24 bytes using the "union" feature
+    children: Vec<SmallVec<[I; 10]>>,
+}
+
+impl<I: Index> Default for NamedHierarchy<I> {
+    fn default() -> Self {
+        Self {
+            // ? NOTE: Since the root has no parent in this context it points to a place outside the vec bounds
+            entities: vec![(I::MAX_VALUE, Name::default())],
+            children: vec![smallvec![]],
+        }
+    }
+}
+
+impl<I: Index> NamedHierarchy<I> {
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    /// Used when the hierarchy must be in a specific order,
+    /// this function takes an vec of entities defined by their parent index
+    /// (on the same vec) and name.
+    ///
+    /// Any root entity should be indexed using `I::MAX_VALUE` or `I::MAX`.
```

Comment:
could you add a const on `NamedHierarchy` to help this?
```rust
    const ROOT_ENTITY: I = I::MAX_VALUE;
```
and then document to use `NamedHierarchy::ROOT_ENTITY`

Reply:
I liked this suggestion, but I used the `NO_PARENT`, the wording of this docs wasn't right. It's not the index of the root entity but the index of the root parent entity